### DISCUSSION TITLE
cfclient Performance Problems When Connected

### DISCUSSION THREAD

**Discussion by DavidHaitch** [bitcraze/discussions#1098] [Q&A]
Has anyone else experienced intense performance problems in cfclient, but only when connected to a drone?
The program starts fine for me, but as soon as I connect to the crazyflie, the whole thing tanks every time. The artificial horizon updates, but doing anything in the UI takes upwards of 30 seconds to register.
Here's the debug info from the About box, captured while experiencing the error. If there are any other logs that would be helpful, I'd be glad to dig them up.

Cfclient
Cfclient version: 2023.11
System: win32
Python: 3.9.10
Qt: 6.6.1
PyQt: 6.6.1

Interface status
radio: Crazyradio version 3.2
UsbCdc: No information available
udp: None
prrt: No information available
cpx: None

Input readers
PySDL2 (0 devices connected)

Input devices
None

Crazyflie
Connected: radio://0/80/2M
Firmware: 4d476a20f811 (CLEAN)

Decks found
bcLedRing: rev=B, adr=0D04106300000080
bcBuzzer: rev=C, adr=0D2E8E57000000AA

Sensors found
BMP388: 1
AK8963: 0
LPS25H: 0

Sensors tests
MPU6500: 0
AK8963: 0
LPS25H: 0

**Comment by ataffanel** [comment]
This is something we have seen before and that is related to graphic performance (I have most often seen it in VMs). The problem is that the main tab and the log graph tab do render graphic quickly and this queues data in the client. This is likely that QT does not support graphic acceleration on your machine. Do you know what graphic adapter your machine has?

To verify that this is indeed the problem you can hide the main flight tab when connected, this should resolve the issue.

One quick hack to get your client working is to reduce the pose update rate, this can be done by changing the 40 in that line by a higher value (for example 400): https://github.com/bitcraze/crazyflie-clients-python/blob/1e888476f1e93fd48bc186bd211b6b88a612ab05/src/cfclient/ui/pose_logger.py#L82. This is the period in ms at which the pose is logged and this controls the update rate of the flight tab.

**Reply by DavidHaitch** [reply]
The adapter is an Intel Iris Xe integrated chipset. 

I changed the poll rate to 200, and that resolved the performance issues. 

Oddly enough, I did not experience these problems in the standard VM, even though it was running on the same laptop.

**Reply by knmcguire** [reply]
Yeah... it is something that we tried to find a way to solve it but unfortunately the crash happens in the QT packaged python library which is written in a different language (c++ perhaps?).  This makes it impossible for us to really figure out what is going wrong... 

We have an issue open for it, but unfortuantely this seems like something that QT should fix: https://github.com/bitcraze/crazyflie-clients-python/issues/656

I had the same issues a lot with QT5 with Win11 but it seems to be 'less' with QT6. But sometimes it still happens...

**Reply by DavidHaitch** [reply]
Would it be possible to allow the refresh rate to be adjusted? Either dynamically in response to detected lag, or explicitly through a setting in software?

**Reply by knmcguire** [reply]
hmmm perhaps! but we need to look into that at one point. I don't know if it would be possible.

If you have any suggestions on how to fix this issue I propose you suggest them in the issue I've given earlier ;) then we can keep it in mind once we come around to fix it.
